SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -346,34 +354,43 @@ Interest InterestFactory::CreatePlayerOwnedActorInterest() const
 
 			if (SpatialGDKSettings->bEnableClientResultTypes)
 			{
-				NewQuery.ResultComponentId = ClientResultType;
+				NewQuery.ResultComponentId = ClientInterestResultType;
 			}
 			else
 			{
 				NewQuery.FullSnapshotResult = true;
 			}
 
-			ClientComponentInterest.Queries.Add(NewQuery);
+			AddComponentQueryPairToInterestComponent(InInterest, ClientEndpointComponentId, NewQuery);
 		}
 	}
+}
 
-	Interest NewInterest;
-	// Server Interest
-	if (SystemConstraints.IsValid() && SpatialGDKSettings->bEnableServerQBI)
-	{
-		NewInterest.ComponentInterestMap.Add(SpatialConstants::POSITION_COMPONENT_ID, ServerComponentInterest);
-	}
-	// Client Interest
-	if (ClientConstraint.IsValid())
+void InterestFactory::AddClientSelfInterest(Interest& InInterest, Worker_EntityId EntityId) const
+{
+	Query NewQuery;
+	// Just an entity ID constraint is fine, as clients should not become authoritative over entities outside their loaded levels
+	NewQuery.Constraint.EntityIdConstraint = EntityId;
+
+	NewQuery.ResultComponentId = ClientAuthResultType;
+
+	AddComponentQueryPairToInterestComponent(InInterest, SpatialConstants::GetClientAuthorityComponent(GetDefault<USpatialGDKSettings>()->bUseRPCRingBuffers), NewQuery);
+}
+
+void InterestFactory::AddComponentQueryPairToInterestComponent(Interest& InInterest, const Worker_ComponentId ComponentId, const Query QueryToAdd)
```

Comment:
```suggestion
void InterestFactory::AddComponentQueryPairToInterestComponent(Interest& InInterest, const Worker_ComponentId ComponentId, const Query& QueryToAdd)
```

Reply:
so, this was done to be explicit that the query will be added as given to the function, and regardless of what happens to the object later the query in the interest map won't change. Is that correct? Happy to change this to a reference regardless, more to check my own understanding